compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -6412,8 +6412,10 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
   def isNonNativeJSClass(sym: Symbol): Boolean =
     !sym.isTrait && isJSType(sym) && !sym.hasAnnotation(JSNativeAnnotation)
 
-  def isNestedJSClass(sym: Symbol): Boolean =
-    sym.isLifted && !sym.originalOwner.isModuleClass && isJSType(sym)
+  def isNestedJSClass(sym: Symbol): Boolean = {
+    sym.isLifted && isJSType(sym) && (
+        !sym.originalOwner.isModuleClass || sym.originalOwner.isLifted)
```

Comment:
Is that third condition even necessary? Could we stop at
```scala
sym.isLifted && isJSType(sym)
```
?

Reply:
IDK. More precisely, I don't know what "lifted" means. For example:

```scala
object A {
  object B
}
```

Is `B` going to be "lifted"?

Reply:
Answer: Yes it is needed. For example, this would cause an assertion error otherwise:

```
object A {
  class Foo extends js.Object
}
```

`Foo` is considered "lifted" but not "nested".

Reply:
By that logic, the following should fail, but doesn't:

```scala
  object TriplyNestedInnerObject_Issue4114 extends js.Object {
    object middle extends js.Object {
      object inner extends js.Object {
        var x = 1
      }
    }
  }
```

Reply:
I have added a test case for this. Full disclosure: I'm mostly experimental coding at this point.

Reply:
Perhaps the `TriplyNestedInnerObject_Issue4114` case doesn't fall because we don't even get to the code path that does that assertion. In that case it's a static module, so there is no `New` for it anywhere.

Reply:
Yes, I guess that is why. Note that this is a real problem, since we are using `isNestedJSClass` to determine how to invoke super calls. (we also use it for constructor exports, but IIUC, that is only an assert/fast-path and could be replaced by looking at the actual parameters).

So I expect code like this to break:

```scala
object A extends js.Object {
  object B extends js.Object {
    class C extends js.Object {
      def foo = 1
    }
  }
}

class D extends A.B.C {
  def foo = super.foo + 1
}
```

Reply:
Strangely, it doesn't fail :S

Reply:
What I do not understand is why this line doesn't take the true branch in the test case:
https://github.com/scala-js/scala-js/blob/v1.1.1/compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala#L675

Reply:
Hmm... It seems that an object inside an object is not lifted whereas a class inside an object is lifted...